From ae636a491a840355c6837800c616ef4a75040dff Mon Sep 17 00:00:00 2001
From: Alberto Murillo <alberto.murillo.silva@intel.com>
Date: Tue, 23 Feb 2016 11:51:18 -0600
Subject: [PATCH] no tty

Only use -tt flag for ssh for privilege escalation via password
since using this flag causes intermitent errors

More details on
	- https://github.com/ansible/ansible/issues/14377
	- https://github.com/ansible/ansible/issues/13876
---
 lib/ansible/plugins/connection/ssh.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ansible/plugins/connection/ssh.py b/lib/ansible/plugins/connection/ssh.py
index 0a0b2bb..0f30313 100644
--- a/lib/ansible/plugins/connection/ssh.py
+++ b/lib/ansible/plugins/connection/ssh.py
@@ -569,7 +569,7 @@ class Connection(ConnectionBase):
         # python interactive-mode but the modules are not compatible with the
         # interactive-mode ("unexpected indent" mainly because of empty lines)
 
-        if in_data:
+        if in_data or not (self._play_context.become and self._play_context.become_pass):            
             cmd = self._build_command('ssh', self.host, cmd)
         else:
             cmd = self._build_command('ssh', '-tt', self.host, cmd)
-- 
2.5.0

